tests/integration/steps/nginx.go

```diff
@@ -0,0 +1,198 @@
+package steps
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/istio/operator/tests/integration/testcontext"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/yaml"
+	"text/template"
+)
+
+//go:embed nginx_config_template.yaml
+var configMapTemplateYaml []byte
+
+// CreateNginxApplication creates a deployment, service  and config map for a nginx application
+func CreateNginxApplication(ctx context.Context, appName, namespace, forwardTo string) (context.Context, error) {
+
+	configMapName := "nginx-conf"
+	ctx, err := createConfigurationCm(ctx, configMapName, namespace, forwardTo)
+	if err != nil {
+		return ctx, err
+	}
+
+	ctx, err = createDeployment(ctx, appName, namespace, configMapName)
+	if err != nil {
+		return ctx, err
+	}
+
+	ctx, err = createService(ctx, appName, namespace, err)
+	if err != nil {
+		return ctx, err
+	}
+
+	return ctx, nil
+}
+
+func createService(ctx context.Context, appName string, namespace string, err error) (context.Context, error) {
+	svc := corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": appName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "http",
+					Port:       80,
```

Comment:
The port could also be dynamic

Reply:
Unless we are making it so because the config map is hardcoded, then I would mention it as a hardcoded value in the feature as well

Reply:
I agree, but I wanted to keep it simple for a starter. I think we can make it dynamic once we see a use case for this, e.g. how we did it with http bin application. Also I think for simplicity we might always want to have it as 80, since this is the most straight forward when doing http requests.

Reply:
I added the the static port to the step